.github/workflows/build.yaml

```diff
@@ -27,26 +27,44 @@ jobs:
     name: ${{ matrix.build.type }} ${{ matrix.build.cxx_compiler }} ${{ matrix.arch }} ${{ matrix.build.os }}
     steps:
       - uses: tenstorrent-metal/metal-workflows/.github/actions/checkout-with-submodule-lfs@v2.0.0
-      - name: Install dependencies
-        if: ${{ matrix.build.runs-on != 'build' }}
-        uses: ./.github/actions/install-metal-deps
-        with:
-          os: ${{ matrix.build.os }}
-      - name: Install dev dependencies
-        if: ${{ matrix.build.runs-on != 'build' }}
-        uses: ./.github/actions/install-metal-dev-deps
-        with:
-          os: ${{ matrix.build.os }}
       - name: Set up dynamic env vars for build
         run: |
           echo "TT_METAL_HOME=$(pwd)" >> $GITHUB_ENV
-      - name: Build C++ libraries and tests
+          echo "RUNNER_UID=$(id -u)" >> $GITHUB_ENV
+          echo "RUNNER_GID=$(id -g)" >> $GITHUB_ENV
+      - name: Determine Docker Image Tag
```

Comment:
this is the third time I've seen this `Determine Docker Image Tag`, think it's time to make our own composite docker action